File: java/1217_MinimumCostToMoveChipsToTheSamePosition.java
class Solution {
    public int minCostToMoveChips(int[] positions) {
        int evens = 0, odds = 0;
        for (int p : positions) {
            if (p % 2 == 0) evens++;
            else odds++;
        }
        return Math.min(odds, evens);
    }
}

File: swift/1217_MinimumCostToMoveChipsToTheSamePosition.swift
class Solution {
    func minCostToMoveChips(_ positions: [Int]) -> Int {
        var evens = 0, odds = 0
        for p in positions {
            if p % 2 == 0 { evens += 1 }
            else { odds += 1 }
        }
        return min(odds, evens)
    }
}
